.circleci/config.yml

```diff
@@ -74,15 +67,17 @@ jobs:
     steps:
     - attach_workspace:
         at: /home/circleci
+    - add_ssh_keys:
+        fingerprints:
+          - "ff:86:1b:40:5e:32:3c:a9:30:cc:96:55:d0:c6:61:ec"
     - checkout
+    - run:
+        name: "Clone docker sample app repo"
+        command: git clone git@github.com:gruntwork-io/sample-app-docker.git "${SAMPLE_APP_DIR}"
```

Comment:
NIT: It would be easier to run the tests locally if the git clone was done inline in the test. It can even create a temporary directory for this purpose that is cleaned out later to ensure the test always runs on a clean version of latest master.

Reply:
I've got `git clone` working inside another repository's tests: https://github.com/robmorgan/terraform-cloudrun-example/blob/master/test/example_test.go#L70, however for my GitHub example I need to push commits back to a GitHub repo. This requires using the SSH private key, which CircleCI does pretty easily out of the box. I could build in more SSH agent logic to my Go helper methods or even add support to Terratest. Thoughts?